server.js

```diff
@@ -144,12 +147,77 @@ storage.initSync();
 
 var _ = require('lodash');    // JavaScript utility library
 var dgram = require('dgram'); // UDP Broadcasting library
-var ips = {activeInterface: 'en0', interfaces: {}};
-if (storage.getItemSync('activeNetworkInterface') !== undefined) {
-    //console.log( storage.getItemSync('activeNetworkInterface'));
-    ips.activeInterface = storage.getItemSync('activeNetworkInterface');
+
+var services = {};
+if (!isMobile) {
+    services.networkInterface = require('network-interfaces');
+}
+services.options = {ipVersion: 4};
+services.ips = {activeInterface: null, tempActiveInterface: null, interfaces: {}};
+services.ip = null
+services.updateAllObjcts = function (ip){
+    console.log("updating all objects with new IP: ", ip);
+    for (let key in objects){
```

Comment:
How does this work with objects on other servers?

Reply:
In case the IP for the server changes, you need to make sure that all objects on the server have the right IP. Because this IP is used to send heartbeats and it's sent out to the editors for other references. You just need to make sure that the Object always has the right IP saved.

Reply:
Oh right, I got confused between this and the userinterface